File: app/src/main/java/com/widyatama/nurseassistant/view/fragment/BottomAddPasienPresenter.java
package com.widyatama.nurseassistant.view.fragment;

import android.content.Context;

import com.widyatama.core.util.SchedulerProviderUtil;
import com.widyatama.nurseassistant.MvpApp;
import com.widyatama.nurseassistant.data.local.sampleDB.dao.PasienDao;
import com.widyatama.nurseassistant.data.model.Pasien;
import com.widyatama.nurseassistant.util.BasePresenter;
import com.widyatama.nurseassistant.view.activity.listPasien.ListPasienViewContract;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

import javax.inject.Inject;

import io.reactivex.disposables.CompositeDisposable;

public class BottomAddPasienPresenter implements BasePresenter<BottomAddPasienViewContract> {

    private Context context;
    private BottomAddPasienViewContract bottomAddPasienViewContract;
    private CompositeDisposable compositeDisposable;
    private SchedulerProviderUtil schedulerProviderUtil;

    @Inject
    PasienDao pasienDao;

    public BottomAddPasienPresenter(Context context, BottomAddPasienViewContract bottomAddPasienViewContract){
        this.context = context;
        this.bottomAddPasienViewContract = bottomAddPasienViewContract;
        MvpApp.appComponent.inject(this);
        compositeDisposable = new CompositeDisposable();
        schedulerProviderUtil = new SchedulerProviderUtil();
    }

    public void addPasien(Pasien pasien){
        ExecutorService exec = Executors.newSingleThreadExecutor();
        exec.execute(new Runnable() {
            @Override
            public void run() {
                pasienDao.insert(pasien);
            }
        });
    }

    @Override
    public void onDetach() {

    }
}

File: app/src/main/java/com/widyatama/nurseassistant/view/fragment/BottomAddPasienPresenter.kt
package com.widyatama.nurseassistant.view.fragment

import com.widyatama.core.base.BasePresenter
import com.widyatama.core.data.session.SessionHelper
import com.widyatama.core.util.SchedulerProviderUtil
import com.widyatama.nurseassistant.data.local.LocalStorageHelper
import com.widyatama.nurseassistant.data.model.Pasien
import com.widyatama.nurseassistant.data.network.ApiHelper
import io.reactivex.disposables.CompositeDisposable
import java.util.concurrent.Executor
import java.util.concurrent.Executors


/**
 * Created by iman on 16/05/2019.
 */

class BottomAddPasienPresenter<V : BottomAddPasienViewContract>
constructor(private val apiHelper: ApiHelper, private val localHelper: LocalStorageHelper,
            sessionHelper: SessionHelper, schedulerProvider: SchedulerProviderUtil, compositeDisposable: CompositeDisposable) : BasePresenter<V>(sessionHelper,
        schedulerProvider, compositeDisposable), BottomAddPasienPresenterContract<V> {

    override fun addPasien(value: Pasien) {
        println("=== insert")
        val exec =Executors.newSingleThreadExecutor()
        exec.execute {
            localHelper.sampleDatabase.pasien().insert(value)
        }
        println("=== done insert")
    }
}